workers/common/src/main/kotlin/common/Extensions.kt

```diff
@@ -47,31 +50,83 @@ inline fun <reified T> ConfigManager.deserializeConfigFileWithDefault(
     defaultPath: String,
     fallbackValue: T,
     context: Context?
+): T = getConfigFileWithDefault<T>(path, defaultPath, fallbackValue, context, ::deserializeConfigFile)
+
+/**
+ * If [path] is not `null`, read the file using the provided [context]. If a [ConfigException] occurs while reading the
+ * file it is rethrown. If [path] is `null`, the file at [defaultPath] is read instead. If the file cannot be read, the
+ * [fallbackValue] is returned.
+ *
+ * This function realizes the contract that if a specific config file is requested, not being able to read it leads to
```

Comment:
As I have stated elsewhere, IMHO we are trying to be too clever here. There are too many options to specify or not specify configuration paths making the whole mechanism hard to understand.

Reply:
For the user there are just two options:
* Specify a path -> Use the file or fail if it does not exist.
* Do not speficy a path -> Use the default file if it exists.